### ISSUE TITLE
model output

### ISSUE THREAD

**Issue by Bicycly23** [adam-osusky/predicting-word-importance#1]
Hello, is the output of the model to determine whether the word is an inserted word or an importance score of the word? thanks.

**Comment by adam-osusky** [comment]
Hello, the model outputs, for each token, logits corresponding to the _“not inserted”_ and _“inserted”_ classes.

**Comment by Bicycly23** [comment]
> Hello, the model outputs, for each token, logits corresponding to the _“not inserted”_ and _“inserted”_ classes.

ok, thank u : )

**Comment by Bicycly23** [comment]
Hello, I would like to use your model as a tool to predict word importance scores, but this is my first time encountering it and I don't know how to convert it. Could you please tell me the method? I also used the methods mentioned in 'Assessing Word Importance Using Models Trained for Semantic Tasks', but I was unable to reproduce the results due to various errors in the environment configuration. Thank you.

> Hello, the model outputs, for each token, logits corresponding to the _“not inserted”_ and _“inserted”_ classes.

**Comment by adam-osusky** [comment]
Our published models behave like a binary token-classifier: every sub-token gets a logit for the classes “inserted” and “not inserted”. If you want a word-level importance score, you can treat P(not inserted) as an “importance” proxy (higher = more important). The snippet below shows one way to do that:
```python
from transformers.models.auto.tokenization_auto import AutoTokenizer
from transformers.models.auto.modeling_auto import AutoModelForTokenClassification
import torch

# load model + tokenizer
name = "adasgaleus/LIM-0.25"
tokenizer = AutoTokenizer.from_pretrained(name)
model = AutoModelForTokenClassification.from_pretrained(name)
model.eval()

joke_txt = """The mathematicians walk into a bar. The first mathematician orders a pint of beer. The second mathematician orders a half. The third, a quarter pint. There is an infinite line up of Mathematicians. The bartender fills up two pints and slides it over to the Mathematicians."""
sentences = [joke_txt]

# tokenize – keep offsets for robust word recovery
enc = tokenizer(
    sentences,
    return_tensors="pt",
    padding=True,
    truncation=True,
    is_split_into_words=False,
    return_offsets_mapping=True,
)

# forward pass
with torch.no_grad():
    logits = model(**{k: v for k, v in enc.items() if k != "offset_mapping"}).logits
    probs = torch.softmax(logits, dim=-1)  # [B, L, 2]

not_inserted_prob_idx = model.config.label2id["not inserted"]
importance = probs[:, :, not_inserted_prob_idx]  # importance = P(not_inserted)

# loop over batch
for b_idx, sent in enumerate(sentences):
    word_ids = enc.word_ids(batch_index=b_idx)
    offsets = enc["offset_mapping"][b_idx]

    # collect per-word spans and scores
    data = {}  # w_id → dict(start, end, scores)
    for tok_idx, w_id in enumerate(word_ids):
        if w_id is None:
            continue  # skip special tokens
        start, end = offsets[tok_idx].tolist()
        score = importance[b_idx, tok_idx].item()

        if w_id in data:
            data[w_id]["end"] = end  # extend span to last sub-token
        else:
            # importance score for a word that is tokenized into more sub-tokens is determined by the first sub-token
            data[w_id] = {"start": start, "end": end, "score": score}

    # build (word, score) list
    word_scores = [(sent[d["start"] : d["end"]], d["score"]) for d in data.values()]

    # sort by importance (descending) and print
    print(f"\nSentence: {sent}")
    for word, score in sorted(word_scores, key=lambda x: x[1], reverse=True):
        print(f"{word:<15} {score:.4f}")
```

But keep in mind, that in our work we were analyzing only the relative ordering of the words by their importance.

All of our models are available here: (https://huggingface.co/collections/adasgaleus/insbert-66819c8e81998a6a08fb9af2)

About the publication “Assessing Word Importance Using Models Trained for Semantic Tasks”, probably the best thing is to reach out directly to its authors in their repository.

**Comment by Bicycly23** [comment]
Thank you very much, I will give it a try : )
